### DISCUSSION TITLE
onClick method doesnt work

### DISCUSSION THREAD

**Discussion by uberautistic** [ImaginativeShohag/Why-Not-Image-Carousel#111] [Q&A] [answered]
`binding.filmsRecycler.carouselListener= object: CarouselListener{
            override fun onCreateViewHolder(
                layoutInflater: LayoutInflater,
                parent: ViewGroup
            ): ViewBinding {
                return FilmRecyclerItemBinding.inflate(layoutInflater, parent, false)
            }
            override fun onBindViewHolder(binding: ViewBinding, item: CarouselItem, position: Int) {
                val currentBinding = binding as FilmRecyclerItemBinding
                currentBinding.imageView.apply {
                    //scaleType= ImageView.ScaleType.CENTER
                    setImage(item)
                }
            }
            override fun onClick(position: Int, carouselItem: CarouselItem) {
                super.onClick(position, carouselItem)
                showToast(requireContext(), position.toString())
            }
        }`
i want to open another activity after click on item but method doesnt even show toast, how can i fix it?

**Comment by ImaginativeShohag** [comment, marked as answer]
If you use a custom layout, then **`carouselListener` will not work.** You should add the click listener to the binding inside `onBindViewHolder`.
